### PR TITLE
Cluster/How to configure APIM for Service Fabric managed cluster.md init

### PR DESCRIPTION
**Pull request by jagilber**
initial release Cluster/How to configure APIM for Service Fabric managed cluster.md

### CODE REVIEW

**Comment by jagilber** [comment]
do not merge. contact sedwards before merging

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 1]
```diff
@@ -0,0 +1,695 @@
+# How to configure APIM Service Fabric Managed Cluster Service Connection
+
+The steps below describe how to configure [Azure API Management](https://learn.microsoft.com/azure/api-management/) (APIM) to route traffic to a back-end service in a Service Fabric managed cluster using PowerShell. For unmanaged clusters, use [Integrate API Management with Service Fabric in Azure](https://learn.microsoft.com/azure/service-fabric/service-fabric-tutorial-deploy-api-management).
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:3]: Product name consistent: Service Fabric Managed Cluster

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+
+## Requirements
+
+- Service Fabric managed cluster deployed using an existing external virtual network. See [Bring your own virtual network](https://learn.microsoft.com/azure/service-fabric/how-to-managed-cluster-networking#bring-your-own-virtual-network) in [Configure network settings for Service Fabric managed clusters](https://learn.microsoft.com/azure/service-fabric/how-to-managed-cluster-networking) for additional  information.
+- [Azure API Management](https://learn.microsoft.com/azure/api-management/).
+- Azure Key vault with certificate.
+
+## Process
+
+1. First, use New-AzResourceGroup to create a resource group to host the virtual network. Run the following code to create a resource group named TestRG in the eastus Azure region.
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:18]: Azure Resource Group
Azure Virtual Network (VNET)
'TestRG'
Azure Region 'EastUS'

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+
+    ```powershell
+    $resourceGroupName = 'testrg'
+    $location = 'eastus'
+    New-AzResourceGroup -Name $resourceGroupName -location $location
+    ```
+
+1. Create a Network Security Group for APIM
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:26]: Azure Network Security Group (NSG)

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+
+    ```powershell
+    $networkSecurityGroup = New-AzNetworkSecurityGroup -Name 'vnet-apim-nsg' -ResourceGroupName $resourceGroupName  -Location $location
+    ```
+
+1. Configure NSG rules for APIM (Management endpoint for Azure portal and PowerShell)
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:32]: for management endpoint in APIM via Azure Portal or PowerShell?

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 5]
```diff
@@ -0,0 +1,695 @@
...
+    Add-AzVirtualNetworkSubnetConfig @apimSubnet
+    ```
+
+1. Then associate the subnets configuration to the virtual network with Set-AzVirtualNetwork.
+
+    ```powershell
+    $virtualNetwork | Set-AzVirtualNetwork
+    ```
+
+1. Prepare the steps for SFMC BYOVNET as per [Bring your own virtual network](https://learn.microsoft.com/azure/service-fabric/how-to-managed-cluster-networking#bring-your-own-virtual-network)
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:92]: Explanation for SFMC BYOVNET?

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 6]
```diff
@@ -0,0 +1,695 @@
...
+
+    - Get the service Id from your subscription for Service Fabric Resource Provider application:
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:94]: ID

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 7]
```diff
@@ -0,0 +1,695 @@
...
+      $sfrpPrincipals = @(Get-AzADServicePrincipal -DisplayName 'Azure Service Fabric Resource Provider')
+      ```
+
+    - Obtain the SubnetId from the existing VNet:
+
+      ```powershell
+      $sfmcSubnetId = ((Get-AzVirtualNetwork -Name $vnet.name -ResourceGroupName $resourceGroupName).Subnets | Where Name -eq $sfmcSubnet.Name | Select Id).Id
+      ```
+
+    - Run the following PowerShell command using the principal ID from previous steps, and assignment scope Id obtained above:
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:106]: ID

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 8]
```diff
@@ -0,0 +1,695 @@
...
+        Sku = 'Standard'
+        AllocationMethod = 'Static'
+        IpAddressVersion = 'IPv4'
+        DomainNameLabel = $domainNameLabel
+    }
+
+    New-AzPublicIpAddress @ip
+    ```
+
+1. Create API Management Service with External VNET integration (It takes around 1 hour)
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:131]: external

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 9]
```diff
@@ -0,0 +1,695 @@
...
+      -Organization $organization `
+      -AdminEmail $adminEmail `
+      -VirtualNetwork $apimNetwork `
+      -VpnType 'External' `
+      -Sku 'Developer' `
+      -PublicIpAddressId $publicIpAddressId
+    ```
+
+1. Create the SFMC within the VNET previously created.
+    > Example existing vnet managed cluster ARM template below: [SFMC ARM deployment template](#sfmc-arm-deployment-template).
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:153]: VNET
managed cluster ARM template?

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 10]
```diff
@@ -0,0 +1,695 @@
...
+
+    > Multiple nodetype example here: [Standard SKU Service Fabric managed cluster, 2 node types, deployed in to existing subnet](https://github.com/Azure-Samples/service-fabric-cluster-templates/tree/master/SF-Managed-Standard-SKU-2-NT-BYOVNET).
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:155]: Service Fabric node type

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 11]
```diff
@@ -0,0 +1,695 @@
...
+1. Configure new Key vault Access policy using a managed identity on existing key vault.
+
+    ```powershell
+    $keyVaultName = 'apimKV'
+    $managedIdentityId = (Get-AzADServicePrincipal -SearchString $apimName).Id
+
+    Set-AzKeyVaultAccessPolicy -VaultName $keyVaultName -ObjectId $managedIdentityId  -PermissionsToSecrets get,list
+    ```
+
+1. Create a keyVault Certificate in APIM
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:206]: Azure Key Vault certificate

File: Cluster/How to configure APIM for Service Fabric managed cluster.md [hunk 12]
```diff
@@ -0,0 +1,695 @@
...
+    ```powershell
+    $kvcertId = 'apimcloud-com'
+    $secretIdentifier = 'https://apimKV.vault.azure.net/secrets/apimcloud-com/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+    $apiMgmtContext = New-AzApiManagementContext -ResourceGroupName $resourceGroupName -ServiceName $apimName
+    
+    $keyvault = New-AzApiManagementKeyVaultObject -SecretIdentifier $secretIdentifier
+    $keyVaultCertificate = New-AzApiManagementCertificate -Context $apiMgmtContext -CertificateId $kvcertId -KeyVault $keyvault
+    ```
+
+1. Create a Service Fabric Backend in APIM using certificate common name (Get the serverX509Name from the Cluster Manifest)
```
Comment (inputoutputcode) [inline Cluster/How to configure APIM for Service Fabric managed cluster.md:217]: backend
common name certificate
get
cluster manifest

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 1 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:3]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 2 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:18]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 3 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:26]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 4 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:32]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 5 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:92]: worded acronym and added additional info under requirements

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 6 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:94]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 8 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:131]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 7 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:106]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 9 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:153]: added more description and updated link name


1. Create the Service Fabric Managed Cluster within the existing VNET previously created above.
    > An example existing VNET Service Fabric Managed Cluster ARM template is provided below: [\sfmc-template.json](#sfmc-arm-deployment-template).

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 10 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:155]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 11 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:206]: fixed

File: Cluster/How to configure APIM for Service Fabric managed cluster.md (hunk 12 above)
Comment (jagilber) [reply Cluster/How to configure APIM for Service Fabric managed cluster.md:217]: fixed

**Pull request closed without merging**